foyer-storage/src/large/flusher.rs

```diff
@@ -280,25 +235,18 @@ where
     V: StorageValue,
     S: HashBuilder + Debug,
 {
-    batch: Batch<K, V, S>,
+    batch: Arc<Mutex<BatchMut<K, V, S>>>,
 
-    rx: mpsc::UnboundedReceiver<Submission<K, V, S>>,
+    notify: Arc<Notify>,
```

Comment:
Replace `Notify` with `CondationVariable` later, for the implementation of `Notify` also includes a mutex.